src/NUnitEngine/nunit.engine.tests/Services/Fakes/FakeProjectService.cs

```diff
@@ -27,7 +27,7 @@ void IProjectService.ExpandProjectPackage(TestPackage package)
             if (_projects.ContainsKey(package.Name))
             {
                 foreach (string assembly in _projects[package.Name])
-                    package.AddSubPackage(new TestPackage(assembly));
+                    package.AddSubPackage(new TestPackage().Named(assembly));
```

Comment:
All these changes make me think we should have a `package.AddSubPackage(tests.dll)` method.

Actually, now I've written that - what would you think of that as a replacement public API for `TestPackage().Named()`? So project extensions would instead call that method, rather than creating individual packages separately and putting them onto the TestPackage?

We might need it to be `var subpackage = package.AddSubPackage("tests.dll")` to make it easy to set settings on the subpackages...

What do you think?

Reply:
One thought - I guess we would lose the ability to create a standalone subpackage. I don't think there's anywhere that needs to do that outside of NUnit's own tests though, is there? I guess it would force us to come up with a nicer solution for testing... 😅

Reply:
I had `AddSubPackage(string)` briefly. But the main use for adding subpackages is in project loader extensions and they need to add settings as well. So if you just add the subpackage, then you have to use an indexer to get it back and add settings. It's easier if they can create the subpackage, add any needed settings and then add it to the original package.

Reply:
We could just have AddSubPackage return the subpackage, couldn't we? 

`var subpackage = package.AddSubPackage("tests.dll")`

Reply:
Good idea. I'll do that!